packages/models/src/resource-binding/middleware.ts

```diff
@@ -0,0 +1,34 @@
+import type { Middleware, RouterContext } from '@ckb-js/kuai-io/lib/types'
+import { computeScriptHash } from '@ckb-lumos/base/lib/utils'
+import { helpers } from '@ckb-lumos/lumos'
+import { BadRequest } from 'http-errors'
+import { ActorName, ActorReference } from '../actor'
+import { ProviderKey } from '../utils'
+
+export function resourceBindingRegisterMiddleware(
+  actorName: ActorName,
+  module: new (...args: Array<unknown>) => unknown,
+): Middleware<RouterContext> {
+  return async (ctx) => {
+    const address = ctx.payload?.params?.address
+    if (!address) return
+
+    try {
+      const lock = helpers.parseAddress(address)
+
+      const lockHash = computeScriptHash(lock)
+      const actorRef = new ActorReference(actorName, `/${lockHash}/`)
+      Reflect.defineMetadata(
```

Comment:
If the `address` is the same, it will call defineMetadata twice, I'm not sure any problem with it.

Reply:
> If the `address` is the same, it will call defineMetadata twice, I'm not sure any problem with it.

The middlewares were deprecated, so I close this conversation.